### PR TITLE
Button Component

### PR DESCRIPTION
**Pull request by aatishpandey**
2022-02-03_19h05_22
Button Component includes:
- Primary Button
- Primary Outline Button 
- Secondary Button
- Secondary Outline Button
- Default Button
- Link Button
- Icon Button
- Icon with Text Button

### CODE REVIEW

File: components/component-stylesheets/button.css [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+.btn-primary-outline:hover{
+    background-color: var(--mr-dark-primary);
+    color: var(--mr-white-text-color);
+}
+
+.btn-link{
+    color: var(--mr-default-color);
+    background-color: transparent;
+    border: none;
+    cursor: pointer;
```
Comment (vivekmehto) [inline components/component-stylesheets/button.css]: its already declared in btn class

File: components/component-stylesheets/button.css [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    padding: 0;
+    font-size: 1.3rem;
+}
+
+/* link button */
+.btn-link-primary{
+    text-decoration: underline;
```
Comment (vivekmehto) [inline components/component-stylesheets/button.css]: Avoid repeating code.

File: components/component-stylesheets/button.css [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    color: var(--mr-light-text-color);
+}
+
+.btn-secondary-outline:hover{
+    background-color:var(--mr-secondary-color);
+}
+
+/* secondary link button */
+.btn-link-secondary{
+    text-decoration: underline;
```
Comment (vivekmehto) [inline components/component-stylesheets/button.css]: same here.

File: components/component-stylesheets/button.css [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+.btn-link-secondary:hover{
+    color: var(--mr-dark-secondary);
+}
+
+/* icon button secondary */
+.btn-icon-secondary{
+    background: none;
+    border: none;
+    font-size: 1.8rem;
```
Comment (vivekmehto) [inline components/component-stylesheets/button.css]: change color only in this class and you can use btn-icon for background border and font-size.

**Review by vivekmehto — COMMENTED** [review]
Avoid repeating code and make reusable classes.

File: components/component-stylesheets/button.css [hunk 5]
```diff
@@ -60,16 +60,16 @@
...
+.btn-icon{
+    background: none;
+    border: none;
+    font-size: 1.8rem;
+    color: var(--mr-primary-color);
+}
 
-    .btn-icon:hover{
-        color: var(--mr-dark-primary);
-    }
```
Comment (vaibhav-mahalle) [inline components/component-stylesheets/button.css]: What is Mr?

**Review by vaibhav-mahalle — COMMENTED** [review]
Looks good

File: components/component-stylesheets/button.css (hunk 5 above)
Comment (vaibhav-mahalle) [reply components/component-stylesheets/button.css]: it is not obvious in the first go

**Review by vaibhav-mahalle — COMMENTED** [review]
Looks good

File: components/component-stylesheets/button.css (hunk 5 above)
Comment (aatishpandey) [reply components/component-stylesheets/button.css]: its naming convention for variables declared for colors in a separate CSS file. I think will not trouble the user, all the variables declared for colors can be seen in the file, Please tell if there is any issue.

**Review by Gautam2305 — COMMENTED** [review]
everything looks fine. Good to merge

File: components/component-stylesheets/button.css (hunk 4 above)
Comment (aatishpandey) [reply components/component-stylesheets/button.css]: thanks for pointing out. fixed in recent commit

**Pull request merged**